aten/src/ATen/autocast_mode.cpp

```diff
@@ -411,6 +411,9 @@ TORCH_LIBRARY_IMPL(aten, Autocast, m) {
                                  std::tuple<Tensor,Tensor,Tensor> (const Tensor&, IntArrayRef, const c10::optional<Tensor>&, const c10::optional<Tensor>&, double),
                                  &ADD_NS(native_layer_norm)>::type::call)));
   KERNEL(ADD_NS(group_norm), "group_norm", Tensor (const Tensor &, int64_t, const c10::optional<Tensor>&, const c10::optional<Tensor>&, double, bool), fp32)
+  KERNEL(ADD_NS(linalg_vector_norm), "linalg_vector_norm", Tensor (const Tensor &, const Scalar&, at::OptionalIntArrayRef, bool, c10::optional<c10::ScalarType>), fp32_set_opt_dtype)
```

Comment:
I think this change is not needed, we shouldn't produce fp32 output in autocast unless user specifically requests it

Reply:
I must have added this because some test was failing. Let me remove it see what happens.

Reply:
If the tests weren't failing before, just redispatch shouldn't have changed it. Adding functions to convert to fp32 list is expensive.

Reply:
This is the failing test: https://github.com/pytorch/pytorch/runs/8049988791?check_suite_focus=true

Reply:
How was it working previously? What caused it to produce fp32 output?

Reply:
Most probably because of these lines:
https://github.com/pytorch/pytorch/blob/ba8080a0cf0511365aef735f30c6e209baceead5/aten/src/ATen/autocast_mode.cpp#L463-L468
and similar lines for `frobenius_norm` and `nuclear_norm`.

Reply:
Ok, the `fp32_set_opt_dtype` makes sense